tests/__tests__/options.test.ts

```diff
@@ -700,6 +701,7 @@ describe('Middleware options', () => {
               },
               1000,
             );
+            store.setState = set;
```

Comment:
Hey @pstrassmann, sorry for the delay. Life has been busy and I haven't made time for working on v3 of zundo; however, I'm here now after some good [discussions in the zustand repo regarding authoring middleware](https://github.com/pmndrs/zustand/discussions/2305). 

Before merging this in, could you add another test case where `store.setState = set` isn't added and we expect `.toHaveBeenCalled(0)`? To put it in English, we should have a test case for zundo, there `wrapTemporal` fails to be called because we haven't modified the store setter.